### DISCUSSION TITLE
Whether cross-compilation is supported for multiple platforms

### DISCUSSION THREAD

**Discussion by TinsFox** [nwutils/nw-builder#981] [General]
I am currently developing and distributing applications using osx arm64 devices, but when my users open the application using osx x64, it prompts that it is not supported. This should be related to the different architecture of the hardware.

So I want to know if it is possible to distribute the installers for all platforms on one machine.

Thank you for your answer.

**Comment by ayushmanchhabra** [comment]
> Whether cross-compilation is supported for multiple platforms

nw-builder does not do any compiling. It only places files in certain locations so that they can be executed by the nw executable.

> when my users open the application using osx x64, it prompts that it is not supported.

Please open a bug report via GitHub issue fir this.

> So I want to know if it is possible to distribute the installers for all platforms on one machine.

Although I have not tried, it should be possible to create a DMG, AppImage or MSIX in MacOS.

**Reply by TinsFox** [reply]
Thanks for your reply, this may be a problem with better-sqlite3 being built on different platforms. But I haven't used nwjs for a while, so I'm not sure what the problem is. But anyway, thank you very much for your work!
